### PR TITLE
Add an error for calling invokeChannel on a dealloc'd FlutterEngine

### PR DESCRIPTION
**Pull request by kevmo314**
*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

See https://github.com/flutter/flutter/issues/128133

Right now, if a FlutterEngine is dealloc'd on iOS, `invokeMethod()` will fail silently. This adds an error message in the event that the binary messenger channel on iOS is not available to make this error case not so silent.

*List which issues are fixed by this PR. You must list at least one issue.*

- https://github.com/flutter/flutter/issues/128133

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by kevmo314** [comment]
I can write some tests for this but would like to get some initial thoughts if this is a worthwhile improvement?

**Comment by stuartmorgan** [comment]
I'll look at this more carefully later, but the main concern here will be making sure that app shutdown doesn't suddenly start throwing a lot of errors; I suspect a lot of plugins rely (perhaps unintentionally) on having this silently no-op.

**Comment by zanderso** [comment]
From PR review triage: It looks like this PR is waiting on a more detailed review from @stuartmorgan.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
I think going from silent no-op to assert is too likely to have bad unintended consequences. A safer approach that would still address the "silently fails" aspect would be to have debug-only logging indicating that the message is being dropped because the engine no longer exists.

**Comment by chinmaygarde** [comment]
Closing this per @stuartmorgan s review.

**Pull request closed without merging**
